File: youtube-links/src/shared/features/links/queries/GetAllPaginatedLinks.ts
import type { SortOrder } from '@/shared/abstractions/SortOrder';

export namespace GetAllPaginatedLinks {
  export interface Query {
    page: number;
    pageSize: number;
    sortColumn: string;
    sortOrder: SortOrder;
    searchTerm: string;
    playlistId: number;
  }
}

File: YoutubeLinks.Shared/Features/Links/Queries/GetAllPaginatedLinks.cs
using FluentValidation;
using MediatR;
using YoutubeLinks.Shared.Abstractions;
using YoutubeLinks.Shared.Features.Links.Responses;

namespace YoutubeLinks.Shared.Features.Links.Queries;

public static class GetAllPaginatedLinks
{
    public class Query : QueryParameters, IRequest<PagedList<LinkDto>>
    {
        public int PlaylistId { get; set; }
    }

    public class Validator : AbstractValidator<Query> { }
}